### ISSUE TITLE
help with inference

### ISSUE THREAD

**Issue by greenmossball** [thomas0809/textreact#3]
hi,

i was able to train the model using one of your scripts but i'm stumbling over how to do inference using my own data. could you share an example on how to do this? it would be greatly appreciated.

**Comment by uranium11010** [comment]
Hi,

Thanks for your interest in our repo.

To do inference, you'll first need to index the nearest neighbors of your test samples. See [Tevatron](https://github.com/thomas0809/tevatron/tree/main), and in particular, [L40-65 of this example script](https://github.com/thomas0809/tevatron/blob/main/scripts/train_RCR.sh#L40) compute the encodings of the chemistry inputs and find the nearest neighbor passages. You'll have to change the test set to your own test set. (If you already have the retrieval model trained and the passages indexed, then you only need to run L40-65. Otherwise, everything before has to be run as well.)

Afterwards, you can run the training script (e.g., [this one](https://github.com/thomas0809/textreact/blob/main/scripts/train_RCR.sh)) but with the `--do_train --do_valid` flags removed. Replace the test set with your own (the path is specified by `--data_path` and `--test_file`).

That should be the high-level. Let me know if you have more questions or if you'd like to have more detailed instructions :)
